components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementAdminService.java

```diff
@@ -427,7 +429,28 @@ public void updateApplication(ServiceProvider serviceProvider) throws IdentityAp
                         " does not have access to the application " + serviceProvider.getApplicationName());
                 throw new IdentityApplicationManagementException("User not authorized");
             }
+            ServiceProvider samlSP = null;
             applicationMgtService = ApplicationManagementService.getInstance();
+            InboundAuthenticationRequestConfig requestConfig =
+                    getInboundAuthenticationRequestConfigByClientType(serviceProvider, SAMLSSO);
+            if (requestConfig != null) {
+                samlSP = applicationMgtService.getServiceProviderByClientId(requestConfig.getInboundAuthKey(),
+                        SAMLSSO , getTenantDomain());
+                if (samlSP != null && isDummyServiceProvider(samlSP.getApplicationName())) {
+                    InboundAuthenticationRequestConfig samlRequestConfig =
+                            getInboundAuthenticationRequestConfigByClientType(samlSP, SAMLSSO);
+                    if (samlRequestConfig != null) {
+                        requestConfig.setProperties(samlRequestConfig.getProperties());
+                    }
+                    //have to delete old application before updating the new application with saml data.Because there
+                    //can't be 2 service Providers with same saml issuer
+                    //delete and update is done using a single method to maintain atomicity
```

Comment:
```suggestion
                    // Required to delete the old application before updating the new application with SAML data. Because there
                    // can't be two service Providers with the same SAML issuer. Therefore
                    // delete and update are done using a single method to maintain atomicity.
```